app/views/exports/pbmc_bank_tubes_content_report.csv.erb

```diff
@@ -20,6 +20,27 @@
 %>
 <%
   rows_array = []
+
+  # Function for finding a racked_tube given a tube and set of tube racks
+  def find_racked_tube_in_tube_racks(tube_racks, tube_id)
+    tube_racks.each do |tube_rack|
+      racked_tube = tube_rack.racked_tubes.find { |rt| rt.tube.id == tube_id }
+      return racked_tube if racked_tube
+    end
+    nil
+  end
+
+  # Fetch the tube racks from the plate
+  tube_racks = @plate.children.select { |child| child.type == 'tube_racks' }.map do |tr|
+    Sequencescape::Api::V2.tube_rack_with_custom_includes(
+      'racked_tubes.tube',
+      nil,
+      barcode: tr.labware_barcode.machine
+    )
+  end
+
+  return if tube_racks.nil?
+
```

Comment:
Just wondering - we _could_ get rid of most of this stuff. We needed to query the racked tubes in that other file you did because the file included the tube rack barcode. This file doesn't actually include it in any of the columns. 

So the only reason to query it here is for this line: `next if racked_tube.nil?`

Not really sure if that line is needed - I don't see how it would be possible to get in the situation where you've got a `downstream_tube` that isn't part if the tube rack. Unless it's from before we make the change to tube racks... in which case, it would be a bonus if this file still works.

Would be much simpler to read without all this stuff - what do you think?

Reply:
Yes I agree, good spot. Makes it much simpler 🙌